src/coreclr/jit/assertionprop.cpp

```diff
@@ -1547,21 +1547,14 @@ AssertionIndex Compiler::optCreateAssertion(GenTree*         op1,
             // Try and see if we can make a subrange assertion.
             if (((assertionKind == OAK_SUBRANGE) || (assertionKind == OAK_EQUAL)))
             {
-                // Keep the casts on small struct fields.
```

Comment:
You can also delete this from the [global propagation](https://github.com/dotnet/runtime/blob/d698901a80227b985ae8e33c7ce138d186bed0ed/src/coreclr/jit/assertionprop.cpp#L2160-L2164) code too, I presume. Though we do not have practically any test coverage for that code.

Reply:
Done, it did not result in any diffs except for when it pushed out some other assertions.